packages/tools/examples/PolySegWasmContourToStackLabelmap/index.ts

```diff
@@ -0,0 +1,302 @@
+label.innerHTML = 'Progress: ';
+
+const progressDetailP = document.createElement('p');
+progressDetailP.id = 'progressDetailP';
+progressDetailP.style.marginLeft = '8px';
+
+inlineContainer.appendChild(label);
+inlineContainer.appendChild(progressDetailP);
+
+// Append the container to the main content
+content.appendChild(inlineContainer);
+
+const viewportGrid = document.createElement('div');
+
+viewportGrid.style.display = 'flex';
+viewportGrid.style.display = 'flex';
+viewportGrid.style.flexDirection = 'row';
+
+const element1 = document.createElement('div');
+const element2 = document.createElement('div');
+element1.style.width = size;
+element1.style.height = size;
+element2.style.width = size;
+element2.style.height = size;
+
+// Disable right click context menu so we can have right click tools
+element1.oncontextmenu = (e) => e.preventDefault();
+element2.oncontextmenu = (e) => e.preventDefault();
+
+viewportGrid.appendChild(element1);
+viewportGrid.appendChild(element2);
+
+content.appendChild(viewportGrid);
+
+createInfoSection(content, { ordered: true })
+  .addInstruction('Draw a contour segmentation on the left viewport')
+  .addInstruction(
+    'Click on the button to convert the contour segmentation to a volume labelmap segmentation'
+  );
+
+// ============================= //
+let toolGroup1, toolGroup2;
+let renderingEngine;
+const toolGroupId1 = 'ToolGroup_Contour';
+const toolGroupId2 = 'ToolGroup_Labelmap';
+const viewportId1 = 'CT_SAGITTAL_CONTOUR';
+const viewportId2 = 'CT_SAGITTAL_LABELMAP';
+
+const segmentIndexes = [1, 2, 3, 4, 5];
+
+addButtonToToolbar({
+  title: 'Convert contour segmentation to labelmap segmentation',
+  onClick: async () => {
+    // add the 3d representation to the 3d toolgroup
+    await segmentation.addSegmentationRepresentations(toolGroupId2, [
+      {
+        segmentationId,
+        type: csToolsEnums.SegmentationRepresentations.Labelmap,
+        options: {
+          polySeg: true,
+        },
+      },
+    ]);
+  },
+});
+
+addDropdownToToolbar({
+  labelText: 'Segment Index',
+  options: { values: segmentIndexes, defaultValue: segmentIndexes[0] },
+  onSelectedValueChange: (number) => {
+    segmentation.segmentIndex.setActiveSegmentIndex(
+      segmentationId,
+      Number(number) as number
+    );
+  },
+});
+
+/**
+ * Runs the demo
+ */
+async function run() {
+  // Init Cornerstone and related libraries
+  await initDemo();
+
+  eventTarget.addEventListener(
+    cornerstoneTools.Enums.Events.POLYSEG_CONVERSION_STARTED,
+    (evt) => {
+      const p = document.getElementById('progressDetailP');
+      p.innerHTML = `Conversion started`;
+    }
+  );
+
+  eventTarget.addEventListener(
+    cornerstoneTools.Enums.Events.POLYSEG_CONVERSION_COMPLETED,
+    (evt) => {
+      const p = document.getElementById('progressDetailP');
+      p.innerHTML = `Conversion completed`;
+    }
+  );
+
+  // Add tools to Cornerstone3D
+  cornerstoneTools.addTool(PanTool);
```

Comment:
Can you use the addManipulationBindings to start removing heavily duplicated code?